src/components/Modals/DeployModal/DeployToWorld_WorldsForEnsOwnersFeature/DeployToWorld.tsx

```diff
@@ -0,0 +1,456 @@
+import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react'
+import classNames from 'classnames'
+import { Button, Field, Icon as DCLIcon, SelectField, Checkbox, Row, Popup, List, DropdownItemProps } from 'decentraland-ui'
+import Modal from 'decentraland-dapps/dist/containers/Modal'
+import { t } from 'decentraland-dapps/dist/modules/translation/utils'
+import { getAnalytics } from 'decentraland-dapps/dist/modules/analytics/utils'
+import { config } from 'config'
+import { isDevelopment } from 'lib/environment'
+import { locations } from 'routing/locations'
+import { Deployment } from 'modules/deployment/types'
+import { FromParam } from 'modules/location/types'
+import CopyToClipboard from 'components/CopyToClipboard/CopyToClipboard'
+import Icon from 'components/Icon'
+import { InfoIcon } from 'components/InfoIcon'
+import { DeployToWorldView, NameType, Props } from './DeployToWorld.types'
+import dclImage from './images/dcl.svg'
+import ensImage from './images/ens.svg'
+
+import styles from './DeployToWorld.module.css'
+
+const EXPLORER_URL = config.get('EXPLORER_URL', '')
+const WORLDS_CONTENT_SERVER_URL = config.get('WORLDS_CONTENT_SERVER', '')
+const CLAIM_NAME_OPTION = 'claim_name_option'
+
+export default function DeployToWorld({
+  name,
+  project,
+  metrics,
+  ensList,
+  externalNames,
+  deployments,
+  isLoading,
+  error,
+  claimedName,
+  onPublish,
+  onRecord,
+  onNavigate,
+  onReplace,
+  onClose,
+  onBack,
+  onFetchExternalNames
+}: Props) {
+  const analytics = getAnalytics()
+
+  const [view, setView] = useState<string>('')
```

Comment:
Would you mind adding tests for this component?

Reply:
It's kind of a big component. It's a pity that original one didn't have tests.

Testing this component will require kind of a lot of time and might break the schedule 😢